srtcore/core.cpp

```diff
@@ -5954,11 +5981,15 @@ int CUDT::receiveMessage(char *data, int len, SRT_MSGCTRL& w_mctrl)
             return res;
     }
 
+    int seqdistance = -1;
+
     if (!m_bSynRecving)
     {
         HLOGC(dlog.Debug, log << CONID() << "receiveMessage: BEGIN ASYNC MODE. Going to extract payload size=" << len);
 
-        int res = m_pRcvBuffer->readMsg(data, len, (w_mctrl));
+        int res = m_pRcvBuffer->readMsg(data, len, (w_mctrl), seqdistance);
```

Comment:
`seqdistance` is hanging for all the following code
It is never changed and could be const.
But it is not needed, if you make `-1` the default value for seqdistance in `m_pRcvBuffer->readMsg(...)`

Reply:
There's predicted a change, which will make the `seqdistance` variable different value. This is simply not implemented here. The `receiveMessage` function will get extra arguments and possibilities to extract packets with controlled sequence numbers.